src/sage/schemes/elliptic_curves/kummer_line.py

```diff
@@ -0,0 +1,1765 @@
+            sage: xP, xQ, xPQ = K(P), K(Q), K(P-Q)
+            sage: X2P, Z2P, XQP, ZQP = K.xDBLADD(*xP.XZ(), *xQ.XZ(), *xPQ.XZ())
+            sage: X2P/Z2P, XQP/ZQP
+            (25, 72)
+            sage: 2*P, P+Q
+            ((25 : 70 : 1), (72 : 27 : 1))
+        """
+        d = self._d
+
+        # In that case, P-Q = (0 : 0 : 1) = T on the curve, so P + Q = 2Q + T
+        # If 2Q = (X : Z), then 2Q + T = (Z : X)
+        if XPQ == 0:
+            X2P, Z2P = self.xDBL(XP, ZP)
+            ZQP, XQP = self.xDBL(XQ, ZQ)
+            return X2P, Z2P, XQP, ZQP
+
+        A = XP + ZP
+        AA = A * A
+        B = XP - ZP
+        BB = B * B
+        E = AA - BB
+        C = XQ + ZQ
+        D = XQ - ZQ
+        DA = D * A
+        CB = C * B
+        t0 = DA + CB
+        t1 = t0 * t0
+        XQP = ZPQ * t1
+        t2 = DA - CB
+        t3 = t2 * t2
+        ZQP = XPQ * t3
+        X2P = AA * BB
+        t4 = d * E
+        t5 = BB + t4
+        Z2P = E * t5
+
+        return X2P, Z2P, XQP, ZQP
+
+
+class KummerLinePoint(SageObject):
+    r"""
+    Kummer point on a Kummer line.
+
+    EXAMPLES::
+
+        sage: E = EllipticCurve(GF(101), [2, 3])
+        sage: K = KummerLine(E)
+        sage: P = E(95, 52)
+
+    A Kummer point can be constructed from `XZ`-coordinates::
+
+        sage: xP = KummerLinePoint(K, [95, 1]); xP
+        (95 : 1)
+
+    Or, it can be constructed from a point on the elliptic curve::
+
+        sage: xP = KummerLinePoint(K, P); xP
+        (95 : 1)
+
+    The point can also be created by calling the Kummer line::
+
+        sage: xP = K(P); xP
+        (95 : 1)
+    """
+
+    def __init__(self, parent, coords):
+        r"""
+        Create a point from the coordinates.
+
+        INPUT:
+
+            - ``parent`` - A Kummer line
+
+            - ``coords`` - either a point P on EllipticCurve or a tuple (X, Z) where `P = (X : * : Z)`; Z is optional
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [2, 3])
+            sage: P = E(95, 52)
+            sage: K = KummerLine(E)
+
+        A Kummer point can be constructed from `XZ`-coordinates::
+
+            sage: xP = KummerLinePoint(K, [95, 1]); xP
+            (95 : 1)
+
+        Z-coordinate is optional, assumed to be 1 then::
+
+            sage: KummerLinePoint(K, 95) == KummerLinePoint(K, [95, 1])
+            True
+            sage: KummerLinePoint(K, 95) == KummerLinePoint(K, [95, 2])
+            False
+            sage: KummerLinePoint(K, 95) == KummerLinePoint(K, [95])
+            True
+
+        Or, it can be constructed from a point on the elliptic curve::
+
+            sage: xP = KummerLinePoint(K, P); xP
+            (95 : 1)
+        """
+        # Ensure the parent is the right type
+        if not isinstance(parent, KummerLine_generic):
+            raise TypeError("not a Kummer line")
+
+        R = parent.base_ring()
+
+        # Point at infinity
+        if coords is None:
+            coords = (R(1), R(0))
```

Comment:
You should use `R.one(), R.zero()` instead, it is faster as it does not go through the coercion model.